web/impact/impact/graphql/query.py

```diff
@@ -7,13 +7,41 @@
 from accelerator.models import (
     EntrepreneurProfile,
     ExpertProfile,
+    UserRole,
 )
-from accelerator_abstract.models.base_user_role import is_finalist_user
 from accelerator_abstract.models.base_user_utils import is_employee
 from graphql import GraphQLError
+
 ENTREPRENEUR_NOT_FOUND_MESSAGE = 'Entrepreneur matching the id does not exist.'
 EXPERT_NOT_FOUND_MESSAGE = 'Expert matching the id does not exist.'
-NON_FINALIST_PROFILE_MESSAGE = 'Sorry, You are not allowed to access this page.'
+NOT_ALLOWED_ACCESS_MESSAGE = 'Sorry, You are not allowed to access this page.'
+
+
+def visible_roles(current_user):
+    basic_user_roles = [
+        UserRole.FINALIST,
+        UserRole.AIR,
+        UserRole.MENTOR,
+        UserRole.PARTNER,
+        UserRole.ALUM
+    ]
+    basic_visible_roles = [UserRole.FINALIST, UserRole.STAFF, UserRole.ALUM]
+
+    current_user_roles = list(
+        current_user.programrolegrant_set.values_list(
```

Comment:
This query does not get current user roles, it gets all user roles that this user holds.

Presumably you could fix this by filtering `user.programrolegrant_set` on program_role.program.program_status

Probably we'd want to write a test about this, also, since it seems like an easy mistake to make. 
test should ensure that a user with a certain user role for an inactive program does not get to see things that only active grant holders should see.